### ISSUE TITLE
Wrong custom image size in multimodal

### ISSUE THREAD

**Issue by KirderfEman** [autogluon/autogluon#2585]
If one sets the image size in the settings under the fit function and timm_image.image_size it changes in the saved config.yaml, so far it seems OK, but if one look at a transformed batch of images from the saved predictor it uses the default timm models config image size not the one set in the predictor.fit() and also seen in the predictors config.
Is this a bug/issue or is there another way that is not in the documentation that you want the user using when need changing the image size?

**Comment by bryanyzhu** [comment]
Hi, can you post more details, like a short code snippet to reproduce the problem? I would like to know how you set the image size in settings under the fit function, thank you.

**Comment by KirderfEman** [comment]
I did it as the documentation said, as you know, e.g. "model.timm_image.image_size": 512. I there another way, did't find it in the doc. nor in the function.
You can then check the results with the below code I created (not from the documentation):

from PIL import Image
import matplotlib.pyplot as plt
file = ##image###
img = Image.open(file)
img = img.convert('RGB')
img = predictor._data_processors['image'][0].train_processor(img)
plt.imshow(img.permute(1, 2, 0))

The fix is to manually overwrite the /multimodal/models/timm_image.py with the code below (not in the doc. nor the function):
        self.config['input_size'] = (3,512, 512)
After the fix the images are in the correct size.

So is it an issue/bug or is there a settings in the fit function that I have missed?

**Comment by yongxinw** [comment]
Hi @KirderfEman, were you trying to load a pretrained `predictor` and use its `_data_processor` to preprocess new images? And you wanted to set a different value of `image_size` by calling the `.fit()` function?

If so, the `image_size` is tied with the `_data_processor`. When you call `.load()` to load a `predictor`, the `predictor` will also restore the `_data_processor`. And then the `.fit()` function first checks if `_data_processor is None`. Only if `_data_processor is None` will the `.fit()` function create a new `_data_processor`. So in your case, since the `_data_processor` is already instantiated in `.load()`, the `.fit()` function would not overwrite it.

Technically you can manually overwrite the `_data_processor` (see `create_fusion_data_processors()`). But since for many models, especially those involving image classification and detection, the model input tensor need to have a fixed size, manual overwrite should be done with caution as it can cause subsequent exceptions for prediction etc.

**Comment by KirderfEman** [comment]
Again:

1. I train a multimodel predictor with the dim size 512,512 by using "model.timm_image.image_size": 512 settings in the fit() function. The model.timm_image.image_size setting is a standard multimodal setting in the config file, so nothing new. 
(If one doesn't set an image size, a default value will be set instead but the image size is an important value that one want control over.)
3. The model is training with x epochs and I can also see in created the config.yaml in the models output folder that the settings is "model.timm_image.image_size": 512.
4. When model is done with the training I load the saved predictor and can via the .train_processor(img) or val_processor(img) , functions used and saved during the training, see that the image size isn't the value 512 I set but the default one from the timm default value for the model. 

Conclusion: The value that one sets and are showed in the models config during the creation of it, isn't used!
I have only used recommended settings for changing the image size but when double check the results, the images size diff. from the one I set and also is showed in the predictors saved config. And I could by more checking see that the image size was the default one from the timm library for the model.

**Comment by sxjscience** [comment]
@KirderfEman we are investigating the issue.

**Comment by sxjscience** [comment]
@KirderfEman Which TIMM backbone are you using?

Actually, a large number of image backbones do not support changing the size of the image.

`timm_image.image_size` means the size we will use if there is no valid image size from TIMM backbones.

**Comment by KirderfEman** [comment]
Yes exactly, most of the transformers have prertrained fixed image sizes, and shouldn't even be useful with bigger image, takes too much of the memory etc. due to the architectures. Here I used the cnn/dnn architectures (eg. effnet) from timm library, and for mostly all those architectures one can change the size and should do it as well, as the default images sizes is for the imagenet training not the own custom dataset, which often need bigger sizes to perform well. 
You can also see that I managed to change to custom image size with manually fix in your codes, if you read the whole post here.

Tips: as for the transformersa arch., write a prompt message for the user, that one cant change size and the default value is uses so the user don't think that the own size set is used, not everyone is aware of the transformer vs dnn setup.

Another suggestion: if the user change to a custom bigger image size than the default, switch to a cnn/dnn SOTA architecture ,as it seems that the user needs a bigger image size for the problem.

But back the issue with changing the image size, hope you have all information to validate and test, please read all the post.

**Comment by sxjscience** [comment]
@KirderfEman thanks for the information and the really detailed explanation! Let me submit a fix for the problem. This is what we overlooked previously.

**Comment by KirderfEman** [comment]
Alright, thanks, please update with the findings.

**Comment by Innixma** [comment]
Adding tracking items, @sxjscience please edit this post to check off items as they are resolved.

- [ ] Fix bug: user-specified fit time image size being ignored at inference time
- [ ] Improve logging: Warn user if they try to change image size for models that don't support it.
- [ ] Improve logging: Add logging of the image size being used during fit (ex: [somewhere in this log](https://auto.gluon.ai/stable/tutorials/multimodal/image_prediction/beginner_image_cls.html#use-automm-to-fit-models)).
- [ ] Stretch: Auto-detect optimal image size based on data, auto-select optimal model based on image size. (Maybe worth creating new issue and marking as potential 0.8 enhancement)

**Comment by sxjscience** [comment]
@KirderfEman PR submitted: https://github.com/autogluon/autogluon/issues/2585

**Comment by sxjscience** [comment]
@KirderfEman The issue should have been fixed. Feel free to try out the new version and reopen it if there are still issues.
